packages/table/src/locator.ts

```diff
@@ -49,17 +49,52 @@ export interface ILocator {
 export class Locator implements ILocator {
     private static CELL_HORIZONTAL_PADDING = 10;
 
+    private grid: Grid;
+
+    private rowHeaderWidth = 0;
+    private columnHeaderHeight = 0;
+
+    private numFrozenRows = 0;
+    private numFrozenColumns = 0;
+
     public constructor(
         private tableElement: HTMLElement,
         private bodyElement: HTMLElement,
-        private grid: Grid,
```

Comment:
The `Grid` `constructor` now accepts only the params that don't change over the course of the instance's life. Everything else is set via explicit setter methods below. (This seemed preferable to a constructor that accepts seven arbitrarily ordered parameters in addition to having setters for the sake of mutation.)